src/Mpdf.php

```diff
@@ -1630,6 +1638,73 @@ private function initFontConfig(array $config)
 		return $config;
 	}
 
+	/**
+	 * Load Font Packages
+	 *
+	 * @param array $config
+	 * @return void
+	 * @throws MpdfException
+	 */
+	protected function initFontRegistry(array $config)
+	{
+		/* Init font package config arrays for the font registry */
+		$fontPackageConfigKeys = ['fonttrans', 'backupSubsFont', 'BMPonly', 'sans_fonts', 'serif_fonts', 'mono_fonts'];
+		foreach ($fontPackageConfigKeys as $fontPackageConfig) {
+			$$fontPackageConfig = [];
+		}
+
+		$fontRegistry = isset($config['fontRegistry']) ? $config['fontRegistry'] : new FontRegistry();
+		$autoloadFontConfig = $fontRegistry->getAutoloadConfigSetting();
+
+		foreach ($fontRegistry->getAll() as $fontPackage) {
+			$this->AddFontDirectory($fontPackage->getDirectory());
+			foreach ($fontPackage->getFonts() as $fontName => $fontData) {
+				if (isset($this->fontdata[$fontName])) {
+					if ($this->debug) {
+						throw new MpdfException(sprintf('Font "%s" already registered', $fontName));
+					}
+
+					continue;
+				}
+
+				if (!empty($fontData['sip-ext'])) {
+					if (!empty($this->backupSIPFont) && $this->debug) {
+						throw new MpdfException(sprintf('Mpdf::backupSIPFont already set to "%s" and font "%s" would override it.', $this->backupSIPFont, $fontName));
+					}
+
+					$this->backupSIPFont = $fontData['sip-ext'];
+				}
+
+				$this->fontdata[$fontName] = $fontData;
+			}
+
+			if (!$autoloadFontConfig) {
+				continue;
+			}
+
+			/* If package includes it, add to the languageToFont registry */
+			$languageToFont = $fontPackage->getLanguageToFont();
+			if ($languageToFont) {
+				$this->languageToFont->add($languageToFont);
+			}
+
+			/* Save data to font package config */
+			$fonttrans[] = $fontPackage->getFontAliases();
+			$backupSubsFont[] = $fontPackage->getBackupSubsFonts();
+			$BMPonly[] = $fontPackage->getBmpFonts();
+			$fontFamilySubstitution = $fontPackage->getFontFamilySubstitution();
+			$sans_fonts[] = isset($fontFamilySubstitution['sans_fonts']) ? $fontFamilySubstitution['sans_fonts'] : [];
+			$serif_fonts[] = isset($fontFamilySubstitution['serif_fonts']) ? $fontFamilySubstitution['serif_fonts'] : [];
+			$mono_fonts[] = isset($fontFamilySubstitution['mono_fonts']) ? $fontFamilySubstitution['mono_fonts'] : [];
+		}
+
+		/* Combine and save font package config to associated Mpdf properties */
+		foreach ($fontPackageConfigKeys as $fontPackageConfig) {
+			$$fontPackageConfig = array_merge([], ...$$fontPackageConfig); // flatten array
```

Comment:
Note: `array_merge` is called exactly 6 times, regardless of whether it's used inside this foreach loop or if this bit of 'magic' is refactored out so [we worked directly with the original variables](https://github.com/mpdf/mpdf/pull/2161/changes#diff-fd95295a9b6e9d877d9735d2e5d3f2f44579d89f77698ea4454ca4a7baffa544R1651).